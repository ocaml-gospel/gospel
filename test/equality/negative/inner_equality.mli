type ('a, 'b) t

val f :
  ('a list -> 'a list -> bool) -> ('a list, 'b) t -> ('a list, 'b) t -> bool
(*@ equality *)
