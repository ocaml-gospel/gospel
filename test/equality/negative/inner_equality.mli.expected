
*******************************
********** Parsed file ********
*******************************
type ('a, 'b) t
val f :
  ('a list -> 'a list -> bool) -> ('a list, 'b) t -> ('a list, 'b) t -> bool
[@@gospel {| equality |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type ('a, 'b) t
  

val f :
  ('a list -> 'a list -> bool) -> ('a list, 'b) t -> ('a list, 'b) t -> bool
(*@ equality
     *)
File "inner_equality.mli", lines 3-5, characters 0-25:
Error: Type checking error: type is incompatible with equality clause
